config/routes.rb

```diff
@@ -221,6 +221,10 @@
     :as => "approve_hat_request"
   post "/hats/reject_request/:id" => "hats#reject_request",
     :as => "reject_hat_request"
+  get "/hats/doff/:id" => "hats#doff",
```

Comment:
Edit this and the next route to `/hats/:id/doff`, it's sort of an action "on" the hat rather than a separate controller.